src/components/TransactionsInterpreter/TransactionsInterpreter.ts

```diff
@@ -27,51 +26,58 @@ export type DenominationConfig = {
 
 export type ParseTransactionsConfiguration = {
   denominationConfig: DenominationConfig;
-  numInitCharactersForScAddress: number;
+  networkAddress?: string;
 };
 
 const defaultConfig: ParseTransactionsConfiguration = {
   denominationConfig: {
-    egldLabel: 'EGLD'
+    egldLabel: mainnetEgldLabel
   },
-  numInitCharactersForScAddress: NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS
+  networkAddress: ''
 };
 
 export function parseTransactions(
   transactions: UITransactionType[],
   address: string,
-  { denominationConfig, numInitCharactersForScAddress } = defaultConfig
+  { denominationConfig, networkAddress } = defaultConfig
 ): ExtendedTransactionType[] {
   return transactions.map((transaction) =>
-    processTransaction(
+    processTransaction({
       transaction,
       address,
       denominationConfig,
-      numInitCharactersForScAddress
-    )
+      networkAddress
+    })
   );
 }
 
-export function processTransaction(
-  transaction: UITransactionType,
-  address: string,
-  denominationConfig: DenominationConfig = {
+type ProcessTransactionParams = {
+  transaction: UITransactionType;
+  address: string;
+  denominationConfig: DenominationConfig;
+  networkAddress?: string;
+};
+
+export function processTransaction({
+  transaction,
+  address,
+  denominationConfig = {
     egldLabel: 'EGLD'
```

Comment:
use egldlabel string from a constant like mainnetEgldLabel (should be defined in constants like mainnetChainId)